sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -582,9 +621,10 @@ def _fit_stochastic(
             # don't stratify in multilabel classification
             should_stratify = is_classifier(self) and self.n_outputs_ == 1
             stratify = y if should_stratify else None
-            X, X_val, y, y_val = train_test_split(
+            X, X_val, y, y_val, sample_weight, _ = train_test_split(
```

Comment:
I think the validation scores also need to be weighted.

We also need to test for `early_stopping=True` when there are sample weights.